server/service/campaigns.go

```diff
@@ -211,6 +211,16 @@ func (svc *Service) NewDistributedQueryCampaignByIdentifiers(ctx context.Context
 		return nil, ctxerr.Wrap(ctx, err, "finding label IDs")
 	}
 
+	// DetectMissingLabels will return the list of labels that are not found in the database
+	// These labels are considered invalid
+	invalidLabels := fleet.DetectMissingLabels(labelMap, labels)
+	if len(invalidLabels) > 0 {
+		setAuthCheckedOnPreAuthErr(ctx)
+		return nil, ctxerr.Wrap(ctx, &fleet.BadRequestError{
+			Message: fmt.Sprintf("[%s] %s", strings.Join(invalidLabels, ", "), fleet.InvalidLabelSpecifiedErrMsg),
```

Comment:
See note on revising the error message. Thinking we can also drop the brackets unless you saw that pattern elsewhere, as the error message reads a little better without them.